SCons/Tool/__init__.py

```diff
@@ -687,8 +687,16 @@ def __call__(self, env, libnode, **kw):
             print("_LibSonameGenerator: libnode=%r" % libnode.get_path())
 
         soname = _call_env_subst(env, '$SONAME', **kw2)
+        version = _call_env_subst(env, '$SOVERSION', **kw2)
```

Comment:
Perhaps check if the values are not equal?
Also dump the values in the error message.